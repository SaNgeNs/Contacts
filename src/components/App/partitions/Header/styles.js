export const styles = {
  wrap: {
    display: 'flex',
    justifyContent: 'center',
    alignContent: 'center',
    padding: '0 20px',
    backgroundColor: '#3399ff',
  },
  list: {
    listStyle: 'none',
    margin: 0,
    padding: 0,
    display: 'flex',
    justifyContent: 'center',
    alignContent: 'center',
  },
  link: {
    padding: '20px',
    display: 'block',
    textDecoration: 'none',
    color: '#fff',
    '&:hover': {
      backgroundColor: '#33ccff',
    },
  },
};

export default styles;
