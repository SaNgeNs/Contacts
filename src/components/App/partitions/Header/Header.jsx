import React, { memo, useEffect } from 'react';
import withStyles from '@material-ui/core/styles/withStyles';
import { NavLink } from 'react-router-dom';
import getMyCards from 'Actions/getMyCards';
import { useDispatch } from 'react-redux';

import styles from './styles';

const activeStyle = {
  backgroundColor: '#33ccff',
  fontWeight: 'bold',
};

export const Header = ({
  classes,  
}) => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getMyCards());
  }, []);

  return (
    <header className={classes.wrap}>
      <ul className={classes.list}>
        <li>
          <NavLink
            className={classes.link}
            exact
            to="/"
            activeStyle={activeStyle}
          >
            Main
          </NavLink>
        </li>

        <li>
          <NavLink
            className={classes.link}
            exact
            to="/favourite"
            activeStyle={activeStyle}
          >
            Favourite
          </NavLink>
        </li>
      </ul>
    </header>
  );
};

export default memo(withStyles(styles)(Header));
