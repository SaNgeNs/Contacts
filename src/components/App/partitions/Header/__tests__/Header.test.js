import React from 'react';
import customRender from 'Mock/customRender';

import Header from '../Header';

describe('<Header />', () => {
  it('should render component', () => {
    const { container, unmount } = customRender(
      <Header />
    );

    expect(container).toMatchSnapshot();
    unmount();
  });
});
