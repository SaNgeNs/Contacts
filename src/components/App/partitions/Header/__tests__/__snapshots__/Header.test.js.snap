// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render component 1`] = `
<div>
  <header
    class="Header-wrap-1"
  >
    <ul
      class="Header-list-2"
    >
      <li>
        <a
          aria-current="page"
          class="Header-link-3 active"
          href="/"
          style="background-color: rgb(51, 204, 255); font-weight: bold;"
        >
          Main
        </a>
      </li>
      <li>
        <a
          class="Header-link-3"
          href="/favourite"
        >
          Favourite
        </a>
      </li>
    </ul>
  </header>
</div>
`;
